modules/Students/studentEnrolment_manage_editProcess.php

```diff
@@ -115,9 +115,9 @@
                         }
 
                         // Add course enrolments for new Roll Group
-                        $data = array('gibbonStudentEnrolmentID' => $gibbonStudentEnrolmentID);
-                        $sql = "INSERT INTO gibbonCourseClassPerson (`gibbonCourseClassID`, `gibbonPersonID`, `role`, `reportable`)
-                                SELECT gibbonCourseClassMap.gibbonCourseClassID, gibbonStudentEnrolment.gibbonPersonID, 'Student', 'Y'
+                        $data = array('gibbonStudentEnrolmentID' => $gibbonStudentEnrolmentID, 'dateEnrolled' => date('Y-m-d'));
+                        $sql = "INSERT INTO gibbonCourseClassPerson (`gibbonCourseClassID`, `gibbonPersonID`, `role`, `dateEnrolled`, `reportable`)
```

Comment:
When a student is enroled, should it set dateUnenrolled to null, just in case it is the second time a student is enrolled in the class (happens, but not very often), and so the old unenroll ends up older than the new enroll?

Reply:
Hmm, yep most of the cases are doing that, using an UPDATE statement. Looks like this case of auto-enrol hasn't been checking for existing enrolment, so I'll add that and set it to reset the enrolment date.